### DISCUSSION TITLE
Clarification on Course Creation Field Requirements

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#230] [Q&A]
Hi Kartheek,

I hope you're doing well!

I wanted to get some clarity on the Course Creation process, specifically regarding the fields used in form configuration and data storage.

For Course Creation, we use the following fields in the form configuration:
```
[  
  "framework",  
  "subjectIds",  
  "topicsIds",  
  "audience",  
  "targetBoardIds",  
  "targetMediumIds",  
  "targetGradeLevelIds",  
  "targetSubjectIds"  
] 
``` 
![Course-From-Config](https://github.com/user-attachments/assets/5326b916-41e1-4d90-b21b-01a2fd4ede4a)

When the course is saved, these fields are stored like this:
```
targetBoardIds: ["pragatifw_board_nios"]  
targetMediumIds: ["pragatifw_medium_english"]  
```
![Course-Create-Update](https://github.com/user-attachments/assets/5fd75f88-2809-4964-94b4-e501d95b97ab)

However, for simple content types like PDFs, H5P, HTML, and Video, we tag fields in a more straightforward manner:
```
board: "NIOS",  
gradeLevel: ["Work Sheets"] 
```
![PDF-content](https://github.com/user-attachments/assets/cf30512a-a0a6-45dc-82d2-6d7600c438a7)

**Content Update Payload:**
![PDF-content-update](https://github.com/user-attachments/assets/b167b1e4-008a-4632-af2e-213d4e9cbe37)

**Question:**
Is it mandatory to store **targetBoardIds**, **targetMediumIds**, **targetGradeLevelIds**, **etc**., for Courses?

If we were to use simpler tags (e.g., **board: "NIOS"**, **gradeLevel: ["Work Sheets"]**) for Courses instead, would it cause any issues in terms of functionality, compatibility, or any other aspect?

Looking forward to your insights!

@pallakartheekreddy @HarishGangula 
@coolbung @itsvick @snehal0904 @rushi-tekdi @siddhi0719 @komalm @vinod303 @baljeetsingh

**Comment by pallakartheekreddy** [comment]
@rajnishdargan 

Please review the PRD below to understand the organization and target frameworks:
[Project Sunbird PRD](https://project-sunbird.atlassian.net/browse/SB-23174)

I am currently checking for the design wiki and will share it for additional information once I have it.
